### DISCUSSION TITLE
Is the C interface complete/please help adapting examples to C?

### DISCUSSION THREAD

**Discussion by wusspuss** [paullouisageneau/libdatachannel#949] [Q&A]
Walking through examples I found I couldn't adapt this chunk of code to C:
```
rtc::Description::Video media("video", rtc::Description::Direction::RecvOnly);
media.addH264Codec(96);
media.setBitrate(3000);
auto track = pc->addTrack(media);
```
The last line seems equivalent to C's `int rtcAddTrack(int pc, const char *mediaDescriptionSdp)`, but how can I generate that sdp string from C, since there seems to be no equivalent to the api exposed by `rtc::Description::Video `?

**Comment by paullouisageneau** [comment]
The equivalent in the C API is `rtcAddTrackEx()`:
https://github.com/paullouisageneau/libdatachannel/blob/a29ccebff1dce7ee5bbee040443e5aab1bbbcb7c/include/rtc/rtc.h#L280

It generates the media description for you from an `rtcTrackInit` struct:
```c
typedef struct {
	rtcDirection direction;
	rtcCodec codec;
	int payloadType;
	uint32_t ssrc;
	const char *mid;
	const char *name;    // optional
	const char *msid;    // optional
	const char *trackId; // optional, track ID used in MSID
} rtcTrackInit;
```

**Reply by wusspuss** [reply]
Thanks! But I still can't see an equivalent to `setBitrate` with it? In fact `capi.cpp` has no mention of `setBitrate` at all..

**Reply by paullouisageneau** [reply]
Indeed, the equivalent of `setBitrate()` to request a specific bitrate is missing from `rtcTrackInit` for now, see https://github.com/paullouisageneau/libdatachannel/issues/916
